staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -453,6 +453,36 @@ type DeleteOptions struct {
 	// foreground.
 	// +optional
 	PropagationPolicy *DeletionPropagation `json:"propagationPolicy,omitempty" protobuf:"varint,4,opt,name=propagationPolicy"`
+
+	// DryRun decides if the request should be completed or not.
+	// +optional
+	DryRun bool `json:"dryRun,omitempty" protobuf:"varint,5,opt,name=dryRun"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CreateOptions may be provided when creating an API object.
+type CreateOptions struct {
+	TypeMeta `json:",inline"`
+
+	// DryRun decides if the request should be completed or not.
+	// +optional
+	DryRun bool `json:"dryRun,omitempty" protobuf:"varint,1,opt,name=dryRun"`
```

Comment:
I think the decision from the call yesterday was to make this an enumerated string.

Reply:
Absolutely, I haven't taken the time to update yet.